src/main/java/dev/dkvl/womutils/WomUtilsPlugin.java

```diff
@@ -205,6 +249,259 @@ public void onResponse(Call call, Response response)
 		});
 	}
 
+	private void sendPlayerRequest(Request request, String target, boolean add)
+	{
+		okHttpClient.newCall(request).enqueue(new Callback()
+		{
+			@Override
+			public void onFailure(Call call, IOException e)
+			{
+				log.warn("Error adding new member {}.", e.getMessage());
+			}
+
+			@Override
+			public void onResponse(Call call, Response response) {
+				try
+				{
+					buildErrorMessage(response, add);
+				}
+				catch (Exception e)
+				{
+					if (add)
+					{
+						groupMembers.add(target);
+					} else {
+						groupMembers.remove(target);
+					}
+				}
+				response.close();
+			}
+		});
+	}
+
+	private void buildErrorMessage(Response response, boolean add) throws IOException {
+		Type messageType = new TypeToken<JsonObject>() {}.getType();
+		String responseBody = response.body().string();
+		JsonObject data = gson.fromJson(responseBody, messageType);
+		String errorMessage = data.get("message").toString().replace("\"", "");
+
+		if (!add && errorMessage.contains("Successfully"))
+		{
+			return;
+		}
+
+		ChatMessageBuilder cmb = new ChatMessageBuilder();
+		cmb.append(ChatColorType.HIGHLIGHT).append(errorMessage);
+
+		chatMessageManager.queue(QueuedMessage.builder()
+				.type(ChatMessageType.CONSOLE)
+				.runeLiteFormattedMessage(cmb.build())
+				.build());
+	}
+
+	private void sendMembersRequest(Request request)
+	{
+		okHttpClient.newCall(request).enqueue(new Callback()
+		{
+			@Override
+			public void onFailure(Call call, IOException e)
+			{
+				log.warn("Error while fetching members list {}.", e.getMessage());
+			}
+
+			@Override
+			public void onResponse(Call call, Response response) throws IOException {
+				groupMembers.clear();
+				Type typeOfArrayList = new TypeToken<ArrayList<GroupMember>>() {}.getType();
+				String responseBody = response.body().string();
+				ArrayList<GroupMember> data = gson.fromJson(responseBody, typeOfArrayList);
+				response.close();
+
+				for (GroupMember gm : data) {
+					groupMembers.add(gm.getUsername());
+				}
+			}
+		});
+	}
+
+	@Subscribe
+	public void onMenuEntryAdded(MenuEntryAdded event)
+	{
+		if (!config.menuOptions())
+		{
+			return;
+		}
+
+		int groupId = WidgetInfo.TO_GROUP(event.getActionParam1());
+		String option = event.getOption();
+		String target = event.getTarget();
+
+		if (groupId == WidgetInfo.FRIENDS_CHAT.getGroupId()
+				|| groupId == WidgetInfo.FRIENDS_LIST.getGroupId())
+		{
+
+			if (!AFTER_OPTIONS.contains(option))
+			{
+				return;
+			}
+
+			final MenuEntry addMember = new MenuEntry();
```

Comment:
```suggestion
			MenuEntry[] entries = client.getMenuEntries();
			entries = Arrays.copyOf(entries, entries.length + 1);
			MenuEntry addMember = entries[entries.length - 1] = new MenuEntry();
```
Then do `client.setMenuEntries(entries)` at the end and you should be able to remove the `insertMenuEntry` and its corresponding apache import.
And you should be able to remove